cpp/tests/ast/transform_tests.cpp

```diff
@@ -92,6 +92,22 @@ TEST_F(TransformTest, BasicAddition)
   cudf::test::expect_columns_equal(expected, result->view(), verbosity);
 }
 
+TEST_F(TransformTest, BasicEquality)
+{
+  auto c_0   = column_wrapper<int32_t>{3, 20, 1, 50};
+  auto c_1   = column_wrapper<int32_t>{3, 7, 1, 0};
+  auto table = cudf::table_view{{c_0, c_1}};
+
+  auto col_ref_0  = cudf::ast::column_reference(0);
+  auto col_ref_1  = cudf::ast::column_reference(1);
+  auto expression = cudf::ast::operation(cudf::ast::ast_operator::EQUAL, col_ref_0, col_ref_1);
```

Comment:
It might be good to add a test of `EQUAL` with null inputs to validate that we didn't accidentally make `EQUAL` and `NULL_EQUAL` semantically equivalent.

Reply:
Yep can't hurt.